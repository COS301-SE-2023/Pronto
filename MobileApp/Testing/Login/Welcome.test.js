import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import WelcomeScreen from "../../screens/Login/Welcome";

describe('WelcomeScreen', () => {
    it('renders correctly', () => {
        const { getByText, getByTestId } = render(<WelcomeScreen />);

        const title = getByText('Make your first step into university easier');
        const description = getByText('Create customised timetables, find your way around campus and get notified of upcoming events');
        const loginButton = getByText('Login');
        const registerButton = getByText('Register');

        expect(title).toBeTruthy();
        expect(description).toBeTruthy();
        expect(loginButton).toBeTruthy();
        expect(registerButton).toBeTruthy();
        expect(getByTestId('image-background')).toBeTruthy();
    });

    it('navigates to Login screen when Login button is pressed', () => {
        const navigationMock = { navigate: jest.fn() };
        const { getByText } = render(<WelcomeScreen navigation={navigationMock} />);
        const loginButton = getByText('Login');

        fireEvent.press(loginButton);

        expect(navigationMock.navigate).toHaveBeenCalledWith('Login');
    });

    it('navigates to Register screen when Register button is pressed', () => {
        const navigationMock = { navigate: jest.fn() };
        const { getByText } = render(<WelcomeScreen navigation={navigationMock} />);
        const registerButton = getByText('Register');

        fireEvent.press(registerButton);

        expect(navigationMock.navigate).toHaveBeenCalledWith('Register');
    });

    it('matches snapshot', () => {
        const { toJSON } = render(<WelcomeScreen />);

        expect(toJSON()).toMatchSnapshot();
    });
});
