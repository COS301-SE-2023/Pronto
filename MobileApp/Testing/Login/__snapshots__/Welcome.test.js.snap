// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "height": 533.6,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 533.6,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="image-background"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#e32f45",
          "fontSize": 30,
          "fontWeight": "bold",
          "paddingBottom": 15,
          "textAlign": "center",
        }
      }
    >
      Make your first step into university easier
    </Text>
    <Text
      style={
        {
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Create customised timetables, find your way around campus and get notified of upcoming events
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingTop": 70,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#e32f45",
          "borderRadius": 10,
          "elevation": 5,
          "marginRight": "4%",
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 15,
          "shadowColor": "#e32f45",
          "shadowOffset": {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 8,
          "width": "48%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderRadius": 10,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 15,
          "width": "48%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
